core/src/main/java/com/linecorp/armeria/server/cors/CorsDecoratorsFactoryFunction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.cors;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.annotation.DecoratorFactoryFunction;
+import com.linecorp.armeria.server.annotation.decorator.CorsDecorator;
+import com.linecorp.armeria.server.annotation.decorator.CorsDecorators;
+
+/**
+ * A factory which creates a {@link CorsService} decorator when more than two {@link CorsDecorator}s are added.
+ */
+public final class CorsDecoratorsFactoryFunction implements DecoratorFactoryFunction<CorsDecorators> {
+
+    /**
+     * Creates a new decorator with the specified {@code parameter}.
+     */
```

Comment:
OK to remove this block because interface's Javadoc will automatically inherited.